src/Symfony/Component/Intl/Data/Generator/LanguageDataGenerator.php

```diff
@@ -162,11 +162,14 @@ protected function generateDataForMeta(BundleEntryReaderInterface $reader, $temp
 
         sort($this->languageCodes);
 
+        $alpha2ToAlpha3 = $this->generateAlpha2ToAlpha3Mapping($metadataBundle);
+
         return [
             'Version' => $rootBundle['Version'],
             'Languages' => $this->languageCodes,
             'Aliases' => array_column(iterator_to_array($metadataBundle['alias']['language']), 'replacement'),
-            'Alpha2ToAlpha3' => $this->generateAlpha2ToAlpha3Mapping($metadataBundle),
+            'Alpha2ToAlpha3' => $alpha2ToAlpha3,
+            'Alpha3ToAlpha2' => array_flip($alpha2ToAlpha3),
```

Comment:
i think we have to decide whether to compile both mappings, compared to a single source of truth and cache the reverse map in-memory. I tend to prefer the latter and keep the resource files small as possible (unless it's a significant perf gain)

Reply:
So, flipping an array of 250 languages may not take long if you only do it once. But it might be a pain if it is inside some loop in the application code.

This also made the implementation much easier.

Reply:
but we do it once :) e.g. `private Countries::getAlpha3ToAlpha2()` which caches the flipped result in a static variable/property

Reply:
This readEntry methods seems to be advanced classes that do caching for you. If we want to take advantage of that I think we need two mappings on disk. It is not only the data, but also the alphabetic ordering. Does not that justify 2 datasets? I think it does.

And it is generated data anyway, the one source of truth is in the ISO bundle.